rskj-core/src/main/java/co/rsk/peg/BridgeUtils.java

```diff
@@ -114,9 +114,13 @@ public static boolean isFreeBridgeTx(org.ethereum.core.Transaction rskTx, long b
         if (receiveAddress == null)
             return false;
 
+        // Temporary assumption: if areBridgeTxsFree() is true then the current federation
+        // must be the genesis federation.
+        // Once the original federation changes, txs are always paid.
         return StringUtils.equals(Hex.toHexString(receiveAddress), PrecompiledContracts.BRIDGE_ADDR) &&
-               blockchainConfig.getConfigForBlock(blockNumber).areBridgeTxsFree() &&
-               rskTx.getSignature() != null &&
-               blockchainConfig.getCommonConstants().getBridgeConstants().getFederatorPublicKeys().contains(co.rsk.bitcoinj.core.BtcECKey.fromPublicOnly(rskTx.getKey().getPubKey()));
+                blockchainConfig.getConfigForBlock(blockNumber).areBridgeTxsFree() &&
+                rskTx.getSignature() != null &&
+                blockchainConfig.getCommonConstants().getBridgeConstants().getGenesisFederation()
+                        .getPublicKeys().contains(co.rsk.bitcoinj.core.BtcECKey.fromPublicOnly(rskTx.getKey().getPubKey()));
```

Comment:
Maybe we can increase readability by extracting some of this return expression variables